tools/Setup.csx

```diff
@@ -196,15 +196,21 @@ class Setup {
                     vscodeExtensionPath = Path.Join(Environment.GetEnvironmentVariable("HOME"), ".vscode/extensions");
                 }
                 IEnumerable<string> extensions = Directory.EnumerateDirectories(vscodeExtensionPath);
-
-                foreach (string extension in extensions)
+                if (extensions.Any())
                 {
-                    if (extension.Contains("ms-vscode.cpptools"))
+                    foreach (string extension in extensions)
                     {
-                        TargetPath = extension;
-                        break;
+                        if (extension.Contains("ms-vscode.cpptools"))
+                        {
+                            TargetPath = extension;
+                            break;
+                        }
                     }
                 }
+                else 
+                {
+                    throw new InvalidOperationException("Unable to find an installation of VS Code C++ Extension.");
+                }
```

Comment:
Isn't this telling you if you failed to find any extension and you need to move line 203 into a `.Where` clause on line 198?